includes/class-newspack-newsletters-subscription.php

```diff
@@ -14,11 +14,31 @@ class Newspack_Newsletters_Subscription {
 
 	const API_NAMESPACE = 'newspack-newsletters/v1';
 
+	const EMAIL_VERIFIED_META    = 'newspack_newsletters_email_verified';
+	const EMAIL_VERIFIED_REQUEST = 'newspack_newsletters_email_verification_request';
+	const EMAIL_VERIFIED_CONFIRM = 'newspack_newsletters_email_verification';
+
+	const WC_ENDPOINT         = 'newsletters';
+	const SUBSCRIPTION_UPDATE = 'newspack_newsletters_subscription';
+
 	/**
 	 * Initialize hooks.
 	 */
 	public static function init() {
 		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+
+		/** User email verification for subscription management. */
+		add_action( 'resetpass_form', [ __CLASS__, 'set_current_user_email_verified' ] );
+		add_action( 'password_reset', [ __CLASS__, 'set_current_user_email_verified' ] );
+		add_action( 'template_redirect', [ __CLASS__, 'process_email_verification_request' ] );
+		add_action( 'template_redirect', [ __CLASS__, 'process_email_verification' ] );
+
+		/** Subscription management through WC's "My Account".  */
+		add_action( 'wp_enqueue_scripts', [ __CLASS__, 'enqueue_scripts' ] );
+		add_filter( 'woocommerce_get_query_vars', [ __CLASS__, 'add_query_var' ] );
```

Comment:
Nice catch, fixed in `f6ad7ffab8887266a8c6c84d83c6ebdc38776e33`. I didn't hook into the `add_query_var` filter because at that point the endpoint has not been registered yet. The `admin_init` hook worked well in my test and should flush without issue by accessing any dashboard page.

Reply:
Why not on `init`, but `admin_init`? With `admin_init` the flush won't happend until an admin logs in, but this is a front-end-facing feature.

Reply:
`init` is not executed at the appropriate time. The page will stay on `404`.

It's safe to assume that `admin_init` will be executed at an appropriate time, even if it's the worst-case scenario of a cron event eventually triggering it.

Reply:
My mistake! I did something wrong while testing. `init` does execute at an appropriate time. 06fac098c8f4d27d22d5b771f844e88140279861 changes it